design/ceph-deprecate-old-osd-types.md

```diff
@@ -0,0 +1,212 @@
+functions as small as possible; the Ceph operator should contain the majority of the logic related
+to Ceph's configuration. Arguments to `ceph-volume` here will be based on the host devices found in
+stage 1 and processing done in stage 2.
+
+For future security, this preparation pod should only need access to the individual disks which
+`ceph-volume` will use to provision Ceph.
+
+#### Stage 4: Start OSDs
+The last step remaining is to start the OSDs previously created by `ceph-volume`. The
+post-preparation device report stored in step 3 will be used to determine which osds to start on any
+given host. The device report should contain the information necessary to determine which flags
+should be passed to `ceph-osd`.
+
+If possible, the existing trend of using init containers instead of a separate Rook binary
+(trend established for Ceph mons, mgrs, MDSes, and RGWs) should be used here with the intent being
+again to keep most of the logic in the operator. If teardown operations are needed, the Kubernetes
+container `PreStop` lifecycle hook should be used if reasonably possible.
+
+
+## Deprecation plan
+The first step in the deprecation plan is to have the new design in-place. Once the new design is
+available, the new design should always be used for any new OSD devices. The overall deprecation
+plan calls for pre-existing user OSDs to be migrated from the newly deprecated OSDs to OSDs using
+the new design.
+
+Regardless of progress on completing this design, users should be informed in the release notes and upgrade guide of
+Rook v1.1 of the intention to deprecate directory-based OSDs and Filestore as a backing store. Once
+the new OSD design is operational in Rook and is the only code path used for new OSDs, the
+deprecation will be official, and that should be reported to users in the release notes and upgrade
+guide. After there is documentation for manually migrating OSDs, we can expect that the following
+version of Rook will officially drop support for legacy OSDs.
+
+It is ideal if users have access to the migration utility at the time of deprecation; however, the
+utility needn't be released at the same time as Rook since it is a separate utility. The utility
+must be released before or at the same time as the Rook release which officially drops support for
+legacy OSDs. The Rook release which officially drops this support should perform a preflight check
+to ensure that no legacy OSDs exist, and if it finds existing legacy OSDs, it should error out with
+a message instructing users to migrate.
+
+### OSD migration
+
+<!-- HALP! I need some help here figuring out how disks will be removed in practice -->
+
+For nodes with only directory-based OSDs, the user will need to attach some number of new disks to
+those nodes and wait for Rook to provision the new disks for use. The directory-based OSDs will be
+migrated by out-ing one directory-based OSD at a time from Ceph and waiting for Ceph to report that
```

Comment:
nit: if it is known that the OSD isn't coming back, then instead of ``ceph osd out osd.NNN`` the user/operator should do ``ceph osd crush rm osd.NNN``.

Reply:
I'm not super familiar with OSD operations, so that's a critical detail for me. Thanks. :)